### PR TITLE
Use new lib-common 'PatchInstance' pattern

### PR DESCRIPTION
**Pull request by abays**
Introduces the use of `lib-common`'s `PatchIntstance()` func and attempts to remove instance-update race condition concerns

### CODE REVIEW

File: controllers/glance_controller.go [hunk 1]
```diff
@@ -111,6 +111,36 @@ func (r *GlanceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
...
+		err := helper.PatchInstance(ctx, instance)
+		if err != nil {
+			_err = err
+			return
+		}
+	}()
+
+	// If we're not deleting this and the service object doesn't have our finalizer, add it.
+	if instance.DeletionTimestamp.IsZero() && controllerutil.AddFinalizer(instance, helper.GetFinalizer()) {
+		return ctrl.Result{}, err
```
Comment (stuggi) [inline controllers/glance_controller.go:141]: same as `return ctrl.Result{}, nil`, so could also return this explicitly.

File: controllers/glance_controller.go [hunk 2]
```diff
@@ -129,9 +159,7 @@ func (r *GlanceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 		instance.Status.Conditions.Init(&cl)
 
 		// Register overall status immediately to have an early feedback e.g. in the cli
-		if err := r.Status().Update(ctx, instance); err != nil {
-			return ctrl.Result{}, err
-		}
+		return ctrl.Result{}, err
```
Comment (stuggi) [inline controllers/glance_controller.go:162]: same as `return ctrl.Result{}, nil`, so could also return this explicitly.

File: controllers/glanceapi_controller.go [hunk 3]
```diff
@@ -107,6 +107,36 @@ func (r *GlanceAPIReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
...
+		err := helper.PatchInstance(ctx, instance)
+		if err != nil {
+			_err = err
+			return
+		}
+	}()
+
+	// If we're not deleting this and the service object doesn't have our finalizer, add it.
+	if instance.DeletionTimestamp.IsZero() && controllerutil.AddFinalizer(instance, helper.GetFinalizer()) {
+		return ctrl.Result{}, err
```
Comment (stuggi) [inline controllers/glanceapi_controller.go:137]: same as `return ctrl.Result{}, nil`, so could also return this explicitly.

File: controllers/glanceapi_controller.go [hunk 4]
```diff
@@ -125,9 +155,7 @@ func (r *GlanceAPIReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 		instance.Status.Conditions.Init(&cl)
 
 		// Register overall status immediately to have an early feedback e.g. in the cli
-		if err := r.Status().Update(ctx, instance); err != nil {
-			return ctrl.Result{}, err
-		}
+		return ctrl.Result{}, err
```
Comment (stuggi) [inline controllers/glanceapi_controller.go:158]: same as `return ctrl.Result{}, nil`, so could also return this explicitly.

**Review by stuggi — COMMENTED** [review]
looks good to me, could just use `return ctrl.Result{}, nil` as err is nil at the places where we return to update the object.

**Comment by abays** [comment]
> looks good to me, could just use `return ctrl.Result{}, nil` as err is nil at the places where we return to update the object.

Whoops, that's an oversight on my part.  I will change that.

File: controllers/glance_controller.go (hunk 1 above)
Comment (abays) [reply controllers/glance_controller.go]: Changed

File: controllers/glance_controller.go (hunk 2 above)
Comment (abays) [reply controllers/glance_controller.go]: Changed

File: controllers/glanceapi_controller.go (hunk 3 above)
Comment (abays) [reply controllers/glanceapi_controller.go]: Changed

File: controllers/glanceapi_controller.go (hunk 4 above)
Comment (abays) [reply controllers/glanceapi_controller.go]: Changed

**Review by fmount — APPROVED** [review]

**Pull request merged**